src/Microsoft.Framework.FileSystemGlobbing/Internal/Patterns/PatternBuilder.cs

```diff
@@ -8,12 +8,24 @@
 
 namespace Microsoft.Framework.FileSystemGlobbing.Internal.Patterns
 {
-    public static class PatternBuilder
+    public class PatternBuilder
     {
         private static readonly char[] _slashes = new[] { '/', '\\' };
         private static readonly char[] _star = new[] { '*' };
 
-        public static IPattern Build(string pattern)
+        public PatternBuilder()
+        {
+            ComparisonType = StringComparison.OrdinalIgnoreCase;
+        }
+
+        public PatternBuilder(StringComparison comparisonType)
```

Comment:
Because it's cleaner. As I choose not to flow the StringComparison during the execution but set up at construction it make sense to ensure all the building component are consistent.